core/src/main/java/org/elasticsearch/index/snapshots/blobstore/BlobStoreIndexShardRepository.java

```diff
@@ -359,38 +359,33 @@ public BlobStoreIndexShardSnapshot loadSnapshot() {
          */
         protected void finalize(List<SnapshotFiles> snapshots, int fileListGeneration, Map<String, BlobMetaData> blobs) {
             BlobStoreIndexShardSnapshots newSnapshots = new BlobStoreIndexShardSnapshots(snapshots);
-            List<String> blobsToDelete = new ArrayList<>();
             // delete old index files first
-            for (String blobName : blobs.keySet()) {
-                // delete old file lists
+            blobs.keySet().forEach(blobName -> {
```

Comment:
can we please stick to the `for` loop. We should be consistent in our loops unless it's really necessary or beneficial